core/reflection.jvm/src/kotlin/reflect/jvm/internal/util.kt

```diff
@@ -195,3 +199,10 @@ internal fun <M : MessageLite, D : CallableDescriptor> deserializeToDescriptor(
     )
     return MemberDeserializer(context).createDescriptor(proto)
 }
+
+@Suppress("UNCHECKED_CAST")
+internal fun KType.asArrayType(): KClass<Array<*>>? = jvmErasure
```

Comment:
I think this function signature is incorrect because the component type can be primitive, and then it would return a value of e.g. `KClass<ByteArray>?`, which is not a subtype of `KClass<Array<*>>`.

Also please add tests on primitive varargs in both cases

Reply:
This seems to work just fine. The Unchecked cast seems to still go through ¯\_(ツ)_/¯ 

But I can change it if you like